reframe/frontend/executors/__init__.py

```diff
@@ -268,6 +308,9 @@ def __init__(self):
         # Task event listeners
         self.task_listeners = []
 
+        # Associated runner
+        self.runner = None
```

Comment:
The only reason for this dependency is just the printing of the retry message. I would rather have the `runner` print a more generic message like "Retrying XX check(s) that have failed before" and leave the policy untouched.

Reply:
Done.